core/src/main/kotlin/org/evomaster/core/search/gene/GeneUtils.kt

```diff
@@ -212,9 +222,21 @@ object GeneUtils {
         val ret =  string.replace("\\", """\\""")
                 .replace("\"", "\\\\\"")
 
-        if (format.isKotlin()) return ret.replace("\$", "\${\'\$\'}")
+        if (format.isKotlin())  return ret.replace("\$", "\${\'\$\'}")
         //ret.replace("\$", "\\\$")
         else return ret
     }
 
-}
\ No newline at end of file
+    fun getMode(mode: String?): EscapeMode{
```

Comment:
can't this be implemented with just `EscapeMode.valueOf(mode)` ?

Reply:
It could. But as far as I understand it, a value that does not match anything will cause an exception to be thrown. I don't think I covered all the possible modes (and I am not sure I want to enforce this strictly). 

Currently, if the mode is something not supported (i.e. EscapeMode.NONE), then the raw string is returned, rather than crashing the system. 

Let me know if you want a different kind of behaviour for this.

Reply:
as the escape mode is not some inputs from the user, but rather an internal detail of EM, then we need to support all modes that we enumerate.